sycl/source/detail/device_binary_image.cpp

```diff
@@ -153,6 +153,7 @@ RTDeviceBinaryImage::getProperty(const char *PropName) const {
 }
 
 void RTDeviceBinaryImage::init(pi_device_binary Bin) {
+  // Bin != nullptr is guaranteed here.
```

Comment:
I thought about this. This method is used once from constructor where Bin is assigned with newly allocated object. I decided that probably assert there has no sence since the only case when Bin is invalid is when "Bin = new ObjectClass" failed. But if it happen, exception will be thrown and Bin usage will be unreachable.

Reply:
Please let me know if you still think assert should be added, I could easily add it there.

Reply:
No, I agree that in such case it's not required.